### PR TITLE
Replaced hidden stream builder calls with hdfsOpenFile()

### PR DESCRIPTION
**Pull request by JasonAlt**
In hadoop 1.8.5, the stream builder functions are no longer declared public in the shared library. Looks like hdfs.h rapped public declarations with __attribute__((visibility("default"))). And in CMakeLists.txt, they set the default -fvisibility=hidden. The stream builder functions didn't make the cut for the public API.

### CODE REVIEW

**Comment by JasonAlt** [comment]
I did not test these changes. They compile, that is as far as I took it.

**Comment by bbockelm** [comment]
Thanks @JasonAlt!

@matyasselmeci / @brianhlin - can you take a look at this?  Particularly concerned we might need to `#ifdef` this a bit to work with the ancient HDFS version we currently run.

**Comment by matyasselmeci** [comment]
Not that ancient, thankfully: we ship hadoop 2.6.0 in OSG 3.4 and 3.5.

**Comment by brianhlin** [comment]
Sorry, this fell off my radar. @edquist please take a look at this sometime this week

**Comment by edquist** [comment]
> Thanks @JasonAlt!
> 
> @matyasselmeci / @brianhlin - can you take a look at this? Particularly concerned we might need to `#ifdef` this a bit to work with the ancient HDFS version we currently run.

Looks like our hadoop carries a patch named `1213-HDFS-9541.-Add-hdfsStreamBuilder-API-to-libhdfs-to-s.patch`

Which leads me to believe this fix is not necessary for our version of hadoop...

Is the point to make gridftp-hdfs work with non-OSG versions of hadoop?

File: src/gridftp_hdfs_recv.c [hunk 1]
```diff
@@ -277,30 +276,12 @@ hdfs_recv(
...
-
-    hdfs_handle->fd = hdfsStreamBuilderBuild(builder);
-    builder = NULL;
+    hdfs_handle->fd = hdfsOpenFile(hdfs_handle->fs, /* hdfsFS fs */
+                                   hdfs_handle->pathname, /* const char *path */
+                                   O_WRONLY, /* int flags */
+                                   0, /* int bufferSize */
+                                   num_replicas, /* short replication */
+                                   num_replicas == 1 ? (tSize)42949672960 : 0); /* tSize blockSize */
     if (!hdfs_handle->fd)
```
Comment (edquist) [inline src/gridftp_hdfs_recv.c:285]: I presume the logic to convert the hdfsStreamBuilder* calls to hdfsOpenFile is all good - one thing that gets my attention is that this `hdfs_handle->fd` had better be a pointer and not actually an int fd.  If it's an int, then i suppose this error check is wrong (then it should be `if (fd < 0)`), but if it's a pointer, then fd is just a misleading name :-p

**Comment by JasonAlt** [comment]
FWIW, the user was setting this up with Cloudera Hadoop CDH 6.3.3. It has Hadoop 3.0.0.

File: src/gridftp_hdfs_recv.c (hunk 1 above)
Comment (JasonAlt) [reply src/gridftp_hdfs_recv.c:285]: The field name 'fd' is misleading. From the example code snippet at https://hadoop.apache.org/docs/r1.2.1/libhdfs.html:

```
    hdfsFile writeFile = hdfsOpenFile(fs, writePath, O_WRONLY|O_CREAT, 0, 0, 0);
    if(!writeFile) {
          fprintf(stderr, "Failed to open %s for writing!\n", writePath);
          exit(-1);
    }
```

'fd' is actually a hdfsFile. Also see that at https://hyunsik.github.io/hdfs-rs/hdfs/native/fn.hdfsOpenFile.html. And that's how 'fd' is declared in [gridftp_hdfs.h](https://github.com/opensciencegrid/gridftp-hdfs/blob/5731ec21568693721601e06349deceea598c87ef/src/gridftp_hdfs.h#L45):

```
hdfsFS                              fs;
```

**Comment by JasonAlt** [comment]
Closing as stale.

**Pull request closed without merging**